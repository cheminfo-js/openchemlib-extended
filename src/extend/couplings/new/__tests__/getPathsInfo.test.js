'use strict';

const OCLE = require('../../../..');
const getPathsInfo = require('../getPathsInfo');

describe('getPathsInfo test propane', () => {
  it('min:1, max:2', () => {
    var molecule = OCLE.Molecule.fromSmiles('CCC');
    molecule.addImplicitHydrogens();

    var pathsInfo = getPathsInfo(molecule, {
      fromLabel: 'H',
      toLabel: 'H',
      minLength: 1,
      maxLength: 3
    });

    expect(pathsInfo).toMatchSnapshot();
  });
});
