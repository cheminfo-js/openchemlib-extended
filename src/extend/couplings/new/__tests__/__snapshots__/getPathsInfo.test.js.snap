// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPathsInfo test propane min:1, max:2 1`] = `
Array [
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 4,
    "allHydrogens": 3,
    "atomicNo": 6,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 3,
      "doubleBonds": 0,
      "singleBonds": 4,
      "totalBonds": 4,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "C",
    "mass": 0,
    "oclID": "eM@Df\`Xb\`RP\\\\Jh",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 1.7320508075688774,
    "y": 0.5,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 4,
    "allHydrogens": 2,
    "atomicNo": 6,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 2,
    "couplings": Array [],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 3,
      "doubleBonds": 0,
      "singleBonds": 4,
      "totalBonds": 4,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "C",
    "mass": 0,
    "oclID": "eM@HzAbJC}IApj\`",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 0.8660254037844387,
    "y": 0,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 4,
    "allHydrogens": 3,
    "atomicNo": 6,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 3,
      "doubleBonds": 0,
      "singleBonds": 4,
      "totalBonds": 4,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "C",
    "mass": 0,
    "oclID": "eM@Df\`Xb\`RP\\\\Jh",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 0,
    "y": 0.5,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          3,
          0,
          4,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "eMABYYeIhOzJBIUJIU@",
          "gC\`DALjYRZhC~bPHeTdRj@",
          "gJPDALjYR[j\`OzI@bURQJh",
        ],
        "to": 4,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          3,
          0,
          5,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "gC\`DALjYRZhCzROtRADjdbUP",
          "gJPDALjYR[j\`OiHQHDRjRIU@",
        ],
        "to": 5,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          3,
          0,
          6,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYjiHOhhQHDRjJIU@",
        ],
        "to": 6,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          3,
          0,
          7,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYj\`OhhQHDRjJIU@",
        ],
        "to": 7,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiKT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 2.3382685902179845,
    "y": 0.85,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          4,
          0,
          3,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "eMABYYeIhOzJBIUJIU@",
          "gC\`DALjYRZhC~bPHeTdRj@",
          "gJPDALjYR[j\`OzI@bURQJh",
        ],
        "to": 3,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          4,
          0,
          5,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "gC\`DALjYRZhCzROtRADjdbUP",
          "gJPDALjYR[j\`OiHQHDRjRIU@",
        ],
        "to": 5,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          4,
          0,
          6,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYjiHOhhQHDRjJIU@",
        ],
        "to": 6,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          4,
          0,
          7,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYj\`OhhQHDRjJIU@",
        ],
        "to": 7,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiKT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 1.3820508075688775,
    "y": 1.106217782649107,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          5,
          0,
          3,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "gC\`DALjYRZhCzROtRADjdbUP",
          "gJPDALjYR[j\`OiHQHDRjRIU@",
        ],
        "to": 3,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          5,
          0,
          4,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "gC\`DALjYRZhCzROtRADjdbUP",
          "gJPDALjYR[j\`OiHQHDRjRIU@",
        ],
        "to": 4,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          5,
          0,
          6,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eMBBYRZCzlRPQJh",
          "gJPDALfYRYjiHOhlQHDRjJIU@",
        ],
        "to": 6,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          5,
          0,
          7,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eMBBYRZCzlRPQJh",
          "gJPDALfYRYj\`OhlQHDRjJIU@",
        ],
        "to": 7,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiKT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 2.0820508075688773,
    "y": -0.10621778264910708,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          6,
          0,
          3,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYjiHOhhQHDRjJIU@",
        ],
        "to": 3,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          6,
          0,
          4,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYjiHOhhQHDRjJIU@",
        ],
        "to": 4,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          6,
          0,
          5,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eMBBYRZCzlRPQJh",
          "gJPDALfYRYjiHOhlQHDRjJIU@",
        ],
        "to": 5,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          6,
          0,
          7,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gJPDALjYRZj\`OilQHDRjRIU@",
        ],
        "to": 7,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          6,
          0,
          8,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMUIA}EgzQ@bUP",
        ],
        "to": 8,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          6,
          0,
          9,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMUIA}EgzQ@bUP",
        ],
        "to": 9,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          6,
          0,
          10,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMUIA}EgzQ@bUP",
        ],
        "to": 10,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiMT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 1.2160254037844387,
    "y": -0.6062177826491071,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          7,
          0,
          3,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYj\`OhhQHDRjJIU@",
        ],
        "to": 3,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          7,
          0,
          4,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "eFBBYcA}T_iHHeT",
          "eMBBYRZCzhRPQJh",
          "gJPDALfYRYj\`OhhQHDRjJIU@",
        ],
        "to": 4,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          7,
          0,
          5,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eMBBYRZCzlRPQJh",
          "gJPDALfYRYj\`OhlQHDRjJIU@",
        ],
        "to": 5,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          7,
          0,
          6,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gJPDALjYRZj\`OilQHDRjRIU@",
        ],
        "to": 6,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          7,
          0,
          8,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMTA}EgzQ@bUP",
        ],
        "to": 8,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          7,
          0,
          9,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMTA}EgzQ@bUP",
        ],
        "to": 9,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          7,
          0,
          10,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMTA}EgzQ@bUP",
        ],
        "to": 10,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiMT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 0.5160254037844387,
    "y": -0.6062177826491071,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          8,
          0,
          6,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMUIA}EgzQ@bUP",
        ],
        "to": 6,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          8,
          0,
          7,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMTA}EgzQ@bUP",
        ],
        "to": 7,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          8,
          0,
          9,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "eM@HzCzl@",
        ],
        "to": 9,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          8,
          0,
          10,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "eM@HzCzl@",
        ],
        "to": 10,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiKT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": -0.6062177826491071,
    "y": 0.85,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          9,
          0,
          6,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMUIA}EgzQ@bUP",
        ],
        "to": 6,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          9,
          0,
          7,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMTA}EgzQ@bUP",
        ],
        "to": 7,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          9,
          0,
          8,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "eM@HzCzl@",
        ],
        "to": 8,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          9,
          0,
          10,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "eM@HzCzl@",
        ],
        "to": 10,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiKT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": -0.35,
    "y": -0.10621778264910708,
    "z": 0,
  },
  Object {
    "abnormalValence": -1,
    "allConnAtoms": 1,
    "allHydrogens": 0,
    "atomicNo": 1,
    "charge": 0,
    "cipParity": 0,
    "color": 0,
    "connAtoms": 1,
    "couplings": Array [
      Object {
        "atoms": Array [
          10,
          0,
          6,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMUIA}EgzQ@bUP",
        ],
        "to": 6,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          10,
          0,
          7,
        ],
        "diaIDto": "gC\`HALiMT@RHDRj@",
        "length": 3,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "gC\`HALiMTA}EgzQ@bUP",
        ],
        "to": 7,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          10,
          0,
          8,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "eM@HzCzl@",
        ],
        "to": 8,
        "torsion": undefined,
      },
      Object {
        "atoms": Array [
          10,
          0,
          9,
        ],
        "diaIDto": "gC\`HALiKT@RHDRj@",
        "length": 2,
        "pathCodes": Array [
          "fH@Ok@",
          "eF@Hp_Q\`",
          "eM@HzCzl@",
        ],
        "to": 9,
        "torsion": undefined,
      },
    ],
    "customLabel": null,
    "extra": Object {
      "aromaticBonds": 0,
      "cnoHybridation": 0,
      "doubleBonds": 0,
      "hydrogenOnAtomicNo": 6,
      "singleBonds": 1,
      "totalBonds": 1,
      "tripleBonds": 0,
    },
    "implicitHydrogens": 0,
    "isAllylic": false,
    "isAromatic": false,
    "isRing": false,
    "isSmallRing": false,
    "isStabilized": false,
    "isStereoCenter": false,
    "label": "H",
    "mass": 0,
    "oclID": "gC\`HALiKT@RHDRj@",
    "radical": 0,
    "ringBondCount": 0,
    "ringSize": 0,
    "x": 0.35,
    "y": 1.106217782649107,
    "z": 0,
  },
]
`;
